examples/spotify-api-example/src/main/java/com/spotify/apollo/example/AlbumResource.java

```diff
@@ -0,0 +1,109 @@
+package com.spotify.apollo.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spotify.apollo.Client;
+import com.spotify.apollo.Request;
+import com.spotify.apollo.RequestContext;
+import com.spotify.apollo.Response;
+import com.spotify.apollo.Status;
+import com.spotify.apollo.example.data.Album;
+import com.spotify.apollo.example.data.Artist;
+import com.spotify.apollo.route.AsyncHandler;
+import com.spotify.apollo.route.Middlewares;
+import com.spotify.apollo.route.Route;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.StringJoiner;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+
+import okio.ByteString;
+
+/**
+ * The album resource demonstrates how to use asynchronous routes in Apollo.
+ */
+public class AlbumResource {
+
+  private static final String SPOTIFY_API = "https://api.spotify.com";
+  private static final String SEARCH_API = SPOTIFY_API + "/v1/search";
+  private static final String ALBUM_API = SPOTIFY_API + "/v1/albums";
+
+  public Stream<Route<AsyncHandler<Response<ByteString>>>> routes() {
+    // The album resource has two routes. Both are using the autoserializer.
+    return Stream.of(
+        Route.async("GET", "/albums/new", this::getAlbums),
+        Route.async("GET", "/albums/hipster", this::getAlbums)
+    )
+        .map(route -> route.withMiddleware(Middlewares::autoSerialize));
+  }
+
+  public CompletionStage<Response<ArrayList<Album>>> getAlbums(RequestContext requestContext) {
+    String uri = requestContext.request().uri();
+    String tag = uri.substring(uri.lastIndexOf("/") + 1);
+
+    // We need to first query the search API, parse the result, then query the album API.
+    Request searchRequest = Request.forUri(SEARCH_API + "?type=album&q=tag%3A" + tag);
+    Client client = requestContext.requestScopedClient();
+    return client
+        .send(searchRequest)
+        .thenComposeAsync(response -> {
+          String ids = parseResponseAlbumIds(response.payload().get().utf8());
+          return client.send(Request.forUri(ALBUM_API + "?ids=" + ids));
+        })
+        .thenApplyAsync(response -> Response.ok()
+            .withPayload(parseAlbumData(response.payload().get().utf8())))
+        .exceptionally(throwable -> Response
+            .forStatus(Status.INTERNAL_SERVER_ERROR.withReasonPhrase("Something went wrong")));
+  }
+
+  /**
+   * Parses an album response from a
+   * <a href="https://developer.spotify.com/web-api/album-endpoints/">Spotify API album query</a>.
+   *
+   * @param json The json response
+   * @return A list of albums with artist information
+   */
+  private ArrayList<Album> parseAlbumData(String json) {
+    ArrayList<Album> albums = new ArrayList<>();
+    try {
+      ObjectMapper mapper = new ObjectMapper();
```

Comment:
I just thought it was a very light weight object without any state so instanciating in reality didn't do anything, and then I just wanted to avoid passing parameters around. But okay, I see that it seems like a best practice to have a single instance of this so I've changed this now.